upup/pkg/fi/cloudup/awstasks/dnszone.go

```diff
@@ -244,34 +244,48 @@ func (_ *DNSZone) RenderTerraform(t *terraform.TerraformTarget, a, e, changes *D
 		glog.Infof("Existing zone %q found; will configure TF to reuse", aws.StringValue(z.HostedZone.Name))
 
 		e.ZoneID = z.HostedZone.Id
-	}
 
-	if z == nil {
-		// Because we expect most users to create their zones externally,
-		// we now block hostedzone creation in terraform.
-		// This lets us perform deeper DNS validation, but also solves the problem
-		// that otherwise we don't know if TF created the hosted zone
-		// (in which case we should output it) or whether it already existed (in which case we should not)
-		// The root problem here is that TF doesn't have a strong notion of an unmanaged resource
-		return fmt.Errorf("Creation of Route53 hosted zones is not supported for terraform")
-		//tf := &terraformRoute53Zone{
-		//	Name: e.Name,
-		//	//Tags:               cloud.BuildTags(e.Name, nil),
-		//}
-		//
-		//tf.Lifecycle = &terraform.Lifecycle{
-		//	PreventDestroy: fi.Bool(true),
-		//}
-		//
-		//return t.RenderResource("aws_route53_zone", *e.Name, tf)
-	} else {
-		// Same problem here also...
+		// If the user specifies dns=private we'll have a non-nil PrivateVPC that specifies the VPC
+		// that should used with the private Route53 zone. If the zone doesn't already know about the
+		// VPC, we add that association.
 		if e.PrivateVPC != nil {
-			return fmt.Errorf("Route53 private hosted zones are not supported for terraform")
+			assocNeeded := true
+			var vpcName string
+			if e.PrivateVPC.ID != nil {
+				vpcName = *e.PrivateVPC.ID
+				for _, vpc := range z.VPCs {
+					if *vpc.VPCId == vpcName {
+						glog.Infof("VPC %q already associated with zone %q", vpcName, aws.StringValue(z.HostedZone.Name))
+						assocNeeded = false
+					}
+				}
+			} else {
+				vpcName = *e.PrivateVPC.Name
+			}
+
+			fmt.Printf("%+v\n", z)
```

Comment:
Is this a debug statement?  No problem leaving them in, but we use `glog.V(2)` or `glog.V(4)` (or even higher) so that they don't appear in users' output